uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/manage-leave-requests.js

```diff
@@ -132,7 +130,10 @@ define([
         classConfirm: 'btn-' + map[action].btnClass,
         msg: map[action].msg,
         onConfirm: function () {
-          return leaveRequest[action]();
+          return leaveRequest[action]()
+            .then(function () {
+              vm.refresh();
```

Comment:
I'm not 100% about this, but it seems `onConfirm` is waiting for a promise. Does `vm.refresh();` returns another promise? Should you return it as part of the chain? Or maybe you need to return the original result?
```
.then(function (actionResponse) {
  vm.refresh();
  return actionResponse;
});
```

Or maybe this does not really need to return anything at all.